scripts/1-fetch/fetch_wikicommons.py

```diff
@@ -0,0 +1,343 @@
+                subcategories.append(subcat_name)
+                
+        LOGGER.info(f"Found {len(subcategories)} subcategories for {category_name}")
+        return subcategories
+        
+    except requests.HTTPError as e:
+        raise shared.QuantifyingException(f"HTTP Error: {e}", 1)
+    except requests.RequestException as e:
+        raise shared.QuantifyingException(f"Request Exception: {e}", 1)
+    except KeyError as e:
+        raise shared.QuantifyingException(f"KeyError: {e}", 1)
+
+
+def recursively_count_category(session, category_name, visited=None):
+    """
+    Recursively count files and pages in a category and its subcategories.
+    
+    Args:
+        session: requests.Session object
+        category_name: Name of the category to count
+        visited: Set of already visited categories (for cycle detection)
+        
+    Returns:
+        dict: Dictionary with 'files' and 'pages' counts
+    """
+    if visited is None:
+        visited = set()
+        
+    if category_name in visited:
+        LOGGER.warning(f"Cycle detected for category: {category_name}")
+        return {"files": 0, "pages": 0}
+        
+    visited.add(category_name)
+    
+    # Get direct counts for this category
+    counts = get_category_info(session, category_name)
+    
+    # Get subcategories and recursively count them
+    subcategories = get_subcategories(session, category_name)
+    
+    for subcat in subcategories:
+        if subcat not in visited:  # Avoid infinite recursion
+            subcat_counts = recursively_count_category(session, subcat, visited.copy())
+            counts["files"] += subcat_counts["files"]
+            counts["pages"] += subcat_counts["pages"]
+    
+    return counts
+
+
+def write_data(args, license_data):
+    """Write the collected data to CSV file."""
+    if not args.enable_save:
+        return args
+
+    # Create data directory for this phase
+    os.makedirs(PATHS["data_phase"], exist_ok=True)
+
+    if len(license_data) < len(CC_LICENSE_CATEGORIES):
+        LOGGER.error("Unable to fetch all records. Aborting.")
+        return args
+
+    with open(FILE1_COUNT, "w", newline="", encoding="utf-8") as file_obj:
+        writer = csv.DictWriter(
+            file_obj, fieldnames=HEADER1_COUNT, dialect="unix"
+        )
+        writer.writeheader()
+        for row in license_data:
+            writer.writerow(row)
+    
+    LOGGER.info(f"Data written to {FILE1_COUNT}")
+    return args
+
+
+def query_wikicommons(args, session):
+    """Query WikiCommons API for CC license data."""
+    license_data = []
+    
+    for category in CC_LICENSE_CATEGORIES:
+        LOGGER.info(f"Processing category: {category}")
+        
+        try:
+            counts = recursively_count_category(session, category)
+            license_data.append({
+                "LICENSE": category,
+                "FILE_COUNT": counts["files"],
+                "PAGE_COUNT": counts["pages"]
+            })
+    
```

Comment:
Hi @najuna-brian 

Is it possible to have a try block without an except clause? I had to add

```javascript
except:
    pass
```
locally to be able to run your code. I think you should re-evaluate this logic and add an appropriate `except` clause